src/app/menu/menu.component.ts

```diff
@@ -49,4 +53,69 @@ export class MenuComponent implements OnInit {
             this._router.navigate(['/login']);
         });
     }
+
+    // Methods calculateInnerClasses, calculateButtonClasses, calculateButtonBackground, calculateOuterClasses
+  /**
+   * Adds inner class depending on the progress
+   * @param  {number}          stage Stage in the application
+   * @param  {string}          className Class which is to be added
+   * @return {boolean}         Return boolean
+   */
+    calculateInnerClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'inner-circle': true} : {'unlocked-inner-circle': true, 'inner-circle': true};
+    }
+
+    calculateButtonClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'but': true} : {'unlocked-btn': true, 'but': true};
+    }
+
+    calculateButtonBackground(stage: number, name: string) {
+      return  (this.checkStageStatus(stage)) ? this.PATH_PREFIX + name + '.png' : this.DEFAULT_PATH;
+    }
+
+    calculateOuterClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'outer-circle': true} : {'unlocked-btn': true, 'outer-circle': true};
+    }
+
+  // Method checkStageStatus
+  /**
+   * Return stage status i.e. whether the stage can be unlocked or not
+   * @param  {number}          stage Stage in the application
+   * @return {boolean}         Return boolean
+   */
+
+    checkStageStatus(stage: number) {
+      return (this.stage >= stage || (this.stage === stage - 1 && this.activity === this.LAST_ACTIVITY))
+    }
+
+  // Method calculateArcClass
+  /**
+   * Return arc status i.e. whether the class should be added to the arc or not
+   * @param  {number}          stage Stage in the application
+   * @param {number}           index Index of the arc i.e. 1st or 2nd or 3rd
+   * @return {boolean}         Return boolean
+   */
+
+    calculateArcClass(stage: number, index: number) {
+      return (this.stage > stage || (this.stage === stage && this.activity >= index + 1)) ? {'completed': true} : {'': true}
+    }
+
+  // Method calculateDisplayedActivity
+  /**
+   * Returns the activity completed in the current stage
+   * @param  {number}          stage Stage in the application
+   * @return {number}         Activity completed
+   */
+
+    calculateDisplayedActivity(stage: number) {
+      if (stage > this.stage)  {
```

Comment:
@lunayach now, say the user registers and so the progress stored in the database is as follows:
stage: 0 and activity: 0.
Now since the first stage starts off with 1 and this function is triggered with 1 as the parameter.
The 1st if  -> if(stage > this.stage) gets evaluated as true and so the activity completed is returned as this.LAST_ACTIVITY, that is 3.

And so, if a user registers for the first time, the first stage will show as 3/3 activities completed.
Please make the required changes.